src/resolver.rs

```diff
@@ -1446,6 +1447,7 @@ fn search_for_path(
     #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
     enum CaveatLevel {
         None,
+        Import,
```

Comment:
I think this is ranked too high. A new import should never be ranked higher than something like a non-importable audit or preferred exemption/unpublished, as that means that unnecessary change to the imports.lock would happen without doing a prune.

Reply:
A fresh import is marked `FreshImport`, this is for existing imports. They previously had `CaveatLevel::None`, so I thought this was appropriate.

Reply:
I've added a bit more elaboration as to what I was going for in my comment above around this being too high of a ranking (tl;dr with the current implementation, this isn't just for existing imports, but also for local wildcard audits with fresh publisher entries, which definitely shouldn't be above these other cases).

I don't think we need a ranking for fully stale imports other than `None`, though I'm open to an argument for where that would be useful.

Reply:
If we don't rank fully stale imports other than `None`, a local wildcard audit won't be preferred over an existing import. So if you already have some imports set up but then add a local wildcard audit, the imports will remain and the local wildcard audit would be unused (and prunable). I think we want the local wildcard audit to be preferred.